core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -268,20 +270,21 @@ public void initializeSnapshot(SnapshotId snapshotId, List<String> indices, Meta
             throw new RepositoryException(this.repositoryName, "cannot create snapshot in a readonly repository");
         }
         try {
-            if (snapshotFormat.exists(snapshotsBlobContainer, snapshotId.getSnapshot()) ||
-                    snapshotLegacyFormat.exists(snapshotsBlobContainer, snapshotId.getSnapshot())) {
-                throw new InvalidSnapshotNameException(snapshotId, "snapshot with such name already exists");
+            final String snapshotName = snapshotId.getName();
+            if (snapshotFormat.exists(snapshotsBlobContainer, snapshotId.blobId()) ||
+                    snapshotLegacyFormat.exists(snapshotsBlobContainer, snapshotName)) {
+                throw new InvalidSnapshotNameException(snapshotId.getSnapshot(), "snapshot with such name already exists");
```

Comment:
is it now possible to write two snapshots with same name? Is there a test for this?